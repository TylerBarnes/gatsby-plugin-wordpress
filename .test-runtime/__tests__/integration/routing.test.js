import fs from "fs-extra"

describe(`[gatsby-plugin-wordpress] routing`, () => {
  test(`Archive template routing is working`, async () => {
    const archiveReport = await fs.readJson(
      `./.test-runtime/WordPress/reports/archive-pages.json`
    )

    expect(archiveReport.rejected.MediaItem).toBeTruthy()
    expect(archiveReport.accepted.MediaItem).toBeFalsy()

    expect(archiveReport.rejected.TypeLimitTest).toBeTruthy()
    expect(archiveReport.accepted.TypeLimitTest).toBeFalsy()

    expect(archiveReport.rejected.TypeLimit0Test).toBeTruthy()
    expect(archiveReport.accepted.TypeLimit0Test).toBeFalsy()

    expect(archiveReport.accepted.Post).toBeTruthy()
    expect(archiveReport.rejected.Post).toBeFalsy()

    expect(archiveReport.accepted.Page).toBeTruthy()
    expect(archiveReport.rejected.Page).toBeFalsy()

    expect(archiveReport.accepted.TeamMember).toBeTruthy()
    expect(archiveReport.rejected.TeamMember).toBeFalsy()

    expect(archiveReport.accepted.Project).toBeTruthy()
    expect(archiveReport.rejected.Project).toBeFalsy()

    expect(archiveReport).toMatchSnapshot()
  })

  test(`Single template routing is working`, async () => {
    const singleReport = await fs.readJson(
      `./.test-runtime/WordPress/reports/single-pages.json`
    )

    expect(singleReport.rejected.MediaItem).toBeTruthy()
    expect(singleReport.accepted.MediaItem).toBeFalsy()

    expect(singleReport.rejected.TypeLimitTest).toBeTruthy()
    expect(singleReport.accepted.TypeLimitTest).toBeFalsy()

    expect(singleReport.rejected.TypeLimit0Test).toBeTruthy()
    expect(singleReport.accepted.TypeLimit0Test).toBeFalsy()

    expect(singleReport.rejected.Category).toBeTruthy()
    expect(singleReport.accepted.Category).toBeFalsy()

    expect(singleReport.rejected.PostFormat).toBeTruthy()
    expect(singleReport.accepted.PostFormat).toBeFalsy()

    expect(singleReport.accepted.Post).toBeTruthy()
    expect(singleReport.rejected.Post).toBeFalsy()

    expect(singleReport.accepted.Page).toBeTruthy()
    expect(singleReport.rejected.Page).toBeFalsy()

    expect(singleReport.accepted.TeamMember).toBeTruthy()
    expect(singleReport.rejected.TeamMember).toBeFalsy()

    expect(singleReport.accepted.Project).toBeTruthy()
    expect(singleReport.rejected.Project).toBeFalsy()

    expect(singleReport.accepted.TranslationFilterTest).toBeTruthy()
    expect(singleReport.rejected.TranslationFilterTest).toBeFalsy()

    expect(singleReport).toMatchSnapshot()
  })
})
