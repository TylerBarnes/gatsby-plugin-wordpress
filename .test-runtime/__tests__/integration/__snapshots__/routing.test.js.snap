// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[gatsby-plugin-wordpress] routing Archive template routing is working 1`] = `
Object {
  "accepted": Object {
    "Page": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/page/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/archive/Page.js",
        "type": "Page",
      },
    ],
    "Post": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/post/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/archive/ContentNode.js",
        "type": "Post",
      },
    ],
    "Project": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/project/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/archive/ContentNode.js",
        "type": "Project",
      },
    ],
    "TeamMember": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/team/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/archive/ContentNode.js",
        "type": "TeamMember",
      },
    ],
    "TranslationFilterTest": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/translation_filter_t/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/archive/ContentNode.js",
        "type": "TranslationFilterTest",
      },
    ],
  },
  "rejected": Object {
    "Category": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/category/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "Category",
      },
    ],
    "MediaItem": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/attachment/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "MediaItem",
      },
    ],
    "PostFormat": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/type/",
        "reason": "This node type doesn't exist in the Gatsby schema.",
        "status": "Rejected",
        "template": null,
        "type": "PostFormat",
      },
    ],
    "Tag": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/tag/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "Tag",
      },
    ],
    "TypeLimit0Test": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/type_limit_0_test/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimit0Test",
      },
    ],
    "TypeLimitTest": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "/type_limit_test/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
    ],
    "User": Array [
      Object {
        "id": null,
        "kind": "archive",
        "path": "users",
        "reason": "Archive routing is disabled for this type in gatsby-config.js.",
        "status": "Rejected",
        "template": null,
        "type": "User",
      },
    ],
  },
  "reportType": "Archive report",
}
`;

exports[`[gatsby-plugin-wordpress] routing Single template routing is working 1`] = `
Object {
  "accepted": Object {
    "Page": Array [
      Object {
        "id": "cG9zdDoxMjc2",
        "kind": "single",
        "path": "/fr/french-home-page/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "Page",
      },
      Object {
        "id": "cG9zdDo1ODY=",
        "kind": "single",
        "path": "/sample-page/child-page/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "Page",
      },
      Object {
        "id": "cG9zdDoxNw==",
        "kind": "single",
        "path": "/members/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "Page",
      },
      Object {
        "id": "cG9zdDoxNg==",
        "kind": "single",
        "path": "/activity/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "Page",
      },
      Object {
        "id": "cG9zdDoy",
        "kind": "single",
        "path": "/sample-page/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "Page",
      },
    ],
    "Post": Array [
      Object {
        "id": "cG9zdDox",
        "kind": "single",
        "path": "/2020/02/25/hello-world/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/Post.js",
        "type": "Post",
      },
    ],
    "Project": Array [
      Object {
        "id": "cG9zdDoxNzI=",
        "kind": "single",
        "path": "/project/wpgraphql/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "Project",
      },
    ],
    "TeamMember": Array [
      Object {
        "id": "cG9zdDo5OA==",
        "kind": "single",
        "path": "/team_member/jason-bahl/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "TeamMember",
      },
    ],
    "TranslationFilterTest": Array [
      Object {
        "id": "cG9zdDoxMzM1",
        "kind": "single",
        "path": "/translation_filter_t/english-page/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "TranslationFilterTest",
      },
      Object {
        "id": "cG9zdDoxMzEx",
        "kind": "single",
        "path": "/fr/translation_filter_t/french-page/",
        "reason": "Page was created with template.",
        "status": "Accepted",
        "template": "./src/wp-templates/single/ContentNode.js",
        "type": "TranslationFilterTest",
      },
    ],
  },
  "rejected": Object {
    "Category": Array [
      Object {
        "id": "dGVybToyMw==",
        "kind": "single",
        "path": "/category/uncategorized-en/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "Category",
      },
      Object {
        "id": "dGVybTox",
        "kind": "single",
        "path": "/fr/category/uncategorized/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "Category",
      },
    ],
    "MediaItem": Array [
      Object {
        "id": "cG9zdDozNjI=",
        "kind": "single",
        "path": "/members/84-1000x1000/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "MediaItem",
      },
      Object {
        "id": "cG9zdDozMTY=",
        "kind": "single",
        "path": "/members/screen-shot-2020-02-27-at-7-40-56-pm/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "MediaItem",
      },
      Object {
        "id": "cG9zdDoyODQ=",
        "kind": "single",
        "path": "/members/screen-shot-2020-02-18-at-7-04-39-am/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "MediaItem",
      },
      Object {
        "id": "cG9zdDoxOTU=",
        "kind": "single",
        "path": "/project/wpgraphql/1024x1024-300x300/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "MediaItem",
      },
      Object {
        "id": "cG9zdDo5OQ==",
        "kind": "single",
        "path": "/team_member/jason-bahl/avatar/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "MediaItem",
      },
    ],
    "PostFormat": Array [
      Object {
        "id": null,
        "kind": "single",
        "path": "/type/",
        "reason": "This node type doesn't exist in the Gatsby schema.",
        "status": "Rejected",
        "template": null,
        "type": "PostFormat",
      },
    ],
    "TypeLimit0Test": Array [
      Object {
        "id": "cG9zdDoxMDY4",
        "kind": "single",
        "path": "/type_limit_0_test/2/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimit0Test",
      },
      Object {
        "id": "cG9zdDoxMDU3",
        "kind": "single",
        "path": "/type_limit_0_test/1/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimit0Test",
      },
    ],
    "TypeLimitTest": Array [
      Object {
        "id": "cG9zdDoxMTY3",
        "kind": "single",
        "path": "/type_limit_test/11/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMTU2",
        "kind": "single",
        "path": "/type_limit_test/10/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMTQ1",
        "kind": "single",
        "path": "/type_limit_test/9/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMTM0",
        "kind": "single",
        "path": "/type_limit_test/8/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMTIz",
        "kind": "single",
        "path": "/type_limit_test/7/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMTEy",
        "kind": "single",
        "path": "/type_limit_test/6/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMTAx",
        "kind": "single",
        "path": "/type_limit_test/5/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMDkw",
        "kind": "single",
        "path": "/type_limit_test/4/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMDc5",
        "kind": "single",
        "path": "/type_limit_test/3/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMDQ2",
        "kind": "single",
        "path": "/type_limit_test/2/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
      Object {
        "id": "cG9zdDoxMDM1",
        "kind": "single",
        "path": "/type_limit_test/1/",
        "reason": "No matching template exists.",
        "status": "Rejected",
        "template": null,
        "type": "TypeLimitTest",
      },
    ],
  },
  "reportType": "Single page report",
}
`;
